File: rhine/src/main/java/com/qingmei2/rhine/http/service/ServiceManager.java
package com.qingmei2.rhine.http.service;


import javax.inject.Inject;
import javax.inject.Singleton;

import lombok.Getter;

/**
 * Created by Glooory on 17/5/15.
 */
@Singleton
public class ServiceManager {

    @Getter
    private UserInfoService userInfoService;

    @Inject
    public ServiceManager(UserInfoService userInfoService) {
        this.userInfoService = userInfoService;
    }

}

File: rhine/src/main/java/com/qingmei2/rhine/http/service/ServiceManager.kt
package com.qingmei2.rhine.http.service


import javax.inject.Inject
import javax.inject.Singleton

import lombok.Getter

/**
 * Created by Glooory on 17/5/15.
 */
@Singleton
class ServiceManager @Inject
constructor(@field:Getter
            private val userInfoService: UserInfoService)
